File: benchmarks/go/objects.go
package main

import "fmt"

type Point struct{ x, y int64 }

func (p *Point) Sum() int64 { return p.x + p.y }

func main() {
	points := []*Point{}
	for i := int64(0); i < 1000000; i++ {
		points = append(points, &Point{x: i, y: i * 2})
	}
	var total int64
	for _, p := range points {
		total += p.Sum()
	}
	fmt.Println(total)
}

File: benchmarks/node/objects.js
class Point {
  constructor(x, y) { this.x = x; this.y = y; }
  sum() { return this.x + this.y; }
}
const points = [];
for (let i = 0; i < 1000000; i++) points.push(new Point(i, i * 2));
let total = 0;
for (const p of points) total += p.sum();
console.log(total);
